tools/lib/h5diff.c

```diff
@@ -662,6 +681,15 @@ h5diff(const char *fname1, const char *fname2, const char *objname1, const char
         }
     }
 
+    if (opts->page_cache > 0) {
+        if (H5Pset_page_buffer_size(fapl2_id, opts->page_cache, 0, 0) < 0) {
+            error_msg("unable to set page buffer cache size for file access\n");
+            h5tools_setstatus(EXIT_FAILURE);
+            ret_value = -1;
+            goto done;
+        }
+    }
+
     if ((file2_id = h5tools_fopen(fname2, H5F_ACC_RDONLY, fapl2_id, (fapl2_id != H5P_DEFAULT), NULL,
```

Comment:
Not sure how much this matters in practice, but since this call to `h5tools_fopen` checks for `fapl2_id != H5P_DEFAULT` and the changes here now always create a fapl for `fapl2_id`, this fopen call will always fail right away when it fails to open the file with the default VOL/VFD, where previously it would have tried others. Unclear if that's still a good idea at this point or not.

Reply:
So I need to create a var here for custom_vol/vfd?

Reply:
Looks like you can use `opts->custom_vol[0] || opts->custom_vol[1]`